src/iam/iam-base.ts

```diff
@@ -169,6 +170,17 @@ export class IAMBase {
         this.setJWT();
     }
 
+    private async checkBalanceForConnection(): Promise<void> {
+        const gasPrice = await this._provider.getGasPrice();
+        const gas = await this._provider.estimateGas(this._transactionOverrides);
+        const balance =
+            (await this._signer?.provider?.getBalance(await this._signer?.getAddress())) || BigNumber.from(0);
+        const fee = gasPrice.mul(gas).mul(2);
```

Comment:
Why `mul(2)`? Just to be on the safe side?

Reply:
Yes, i did the same calculation as Dmitry did here: https://github.com/energywebfoundation/iam-client-lib/blob/4f2ec9ac05316c89d878a7c35568bd68cc3c5424/src/staking/index.ts#L178

Reply:
Ah gotcha. I think it would be good to have the same comment then
```suggestion
        // multiplier 2 chosen arbitrarily because it is not known how reasonably to choose it (as done in iam-client-lib)
        const fee = gasPrice.mul(gas).mul(2);
```